apps/web/lib/withNonce.tsx

```diff
@@ -0,0 +1,82 @@
+import { GetServerSideProps, GetServerSidePropsContext, GetServerSidePropsResult } from "next";
+import { csp } from "pages/_document";
+import { z } from "zod";
+
+export type WithNonceProps = {
+  nonce?: string;
+};
+
+function getCspPolicy(nonce: string) {
+  const isProduction = process.env.NODE_ENV === "production";
+  //TODO: Do we need to explicitly define it in turbo.json
+  const CSP_POLICY = process.env.CSP_POLICY;
+
+  // Note: "non-strict" policy only allows inline styles otherwise it's the same as "strict"
+  // We can remove 'unsafe-inline' from style-src when we add nonces to all style tags
+  // Maybe see how @next-safe/middleware does it if it's supported.
+  const useNonStrictPolicy = CSP_POLICY === "non-strict-report-only" || CSP_POLICY === "non-strict";
+  if (!useNonStrictPolicy && isProduction) {
+    throw new Error(
+      "Strict CSP policy(for style-src) is not yet supported in production. You can experiment with it in Dev Mode"
+    );
+  }
+  return `
+	default-src 'self' ${isProduction ? "" : "data:"};
+	script-src 'self'  ${
+    isProduction
+      ? "'nonce-" + nonce + "' 'strict-dynamic' 'unsafe-inline' https:"
+      : // Note: We could use 'strict-dynamic' with 'nonce-..' instead of unsafe-inline but in dev mode there are some streaming related scripts that get blocked(because they don't have nonce on them)
```

Comment:
Found that such scripts are there for production build as well. The error doesn't come always and when it comes there is no visible functionality issue. So, we are okay with it for now.
Still keeping unsafe-inline for dev mode so that Next.js doesn't;t show that Error modal box that hinders browsing.